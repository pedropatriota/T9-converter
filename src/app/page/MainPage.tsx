import { Button, Input, MessageBox } from "../components";
import { Template } from "../Template";
import { MainPageHelper } from "./MainPage.helper";
import * as S from "./styles";

export const MainPage: React.FC = () => {
  const {
    value,
    disabled,
    handleInputChange,
    handleClearField,
    inputRef,
    sendPost,
    converted,
  } = MainPageHelper();

  return (
    <Template>
      <S.Container>
        <S.InputContainer>
          <Input
            ref={inputRef}
            value={value}
            handleInputChange={handleInputChange}
            handleClearField={handleClearField}
            placeholder="type a number..."
          />
          <Button
            disabled={disabled}
            onClick={() => sendPost({ number: value })}
          >
            Convert
          </Button>
        </S.InputContainer>
        {!value ? (
          <MessageBox
            title="Waiting for a number..."
            text="Type a number to convert its value in T9."
          />
        ) : converted.length ? (
          <S.ResultContainer data-testid="converted-value-box">
            {converted.map(item => (
              <span key={item}>{`${item}, `}</span>
            ))}
          </S.ResultContainer>
        ) : undefined}
      </S.Container>
    </Template>
  );
};
