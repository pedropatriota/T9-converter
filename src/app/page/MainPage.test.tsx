import "jest-styled-components";
import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { MainPage } from "./MainPage";

describe("MainPage component", () => {
  it("Should respect the initial render of page", () => {
    const { container } = render(<MainPage />);

    const input = screen.getByRole("textbox");
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue("");
    expect(input).toHaveFocus();

    const button = screen.getByRole("button");
    expect(button).toBeDisabled();

    const messageBox = screen.getByTestId("message-box");
    expect(messageBox).toBeInTheDocument();

    const clearIcon = screen.queryByTestId("clear-icon");
    expect(clearIcon).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("Should respect the change in the screen when a user type a number", async () => {
    const { container } = render(<MainPage />);

    const input = screen.getByRole("textbox");
    const button = screen.getByRole("button");
    const messageBox = screen.queryByTestId("message-box");

    userEvent.type(input, "23");
    expect(input).toHaveValue("23");
    expect(button).toBeEnabled();
    expect(messageBox).not.toBeInTheDocument();

    const clearIcon = await screen.findByTestId("clear-icon");
    expect(clearIcon).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("Should respect the change in the screen when a user clear the input", async () => {
    const { container } = render(<MainPage />);

    const input = screen.getByRole("textbox");
    const button = screen.getByRole("button");
    const messageBox = screen.queryByTestId("message-box");

    userEvent.type(input, "23");
    expect(input).toHaveValue("23");
    expect(button).toBeEnabled();
    expect(messageBox).not.toBeInTheDocument();

    const clearIcon = await screen.findByTestId("clear-icon");
    expect(clearIcon).toBeInTheDocument();

    userEvent.click(clearIcon);
    expect(input).toHaveValue("");
    expect(button).toBeDisabled();
    expect(messageBox).not.toBeInTheDocument();
    expect(clearIcon).not.toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it("Should respect the change in the screen when a user request a Post", async () => {
    const { container } = render(<MainPage />);

    const input = screen.getByRole("textbox");
    const button = screen.getByRole("button");
    const messageBox = screen.queryByTestId("message-box");

    userEvent.type(input, "23");
    expect(input).toHaveValue("23");
    expect(button).toBeEnabled();
    expect(messageBox).not.toBeInTheDocument();

    const clearIcon = await screen.findByTestId("clear-icon");
    expect(clearIcon).toBeInTheDocument();

    userEvent.click(button);
    const convertedValueBox = await screen.findByTestId("converted-value-box");
    expect(convertedValueBox).toBeInTheDocument();
    expect(convertedValueBox).toHaveTextContent(
      "ad, ae, af, bd, be, bf, cd, ce, cf,"
    );

    expect(container).toMatchSnapshot();
  });
});
