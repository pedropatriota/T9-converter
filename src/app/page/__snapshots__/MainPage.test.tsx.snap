// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component Should respect the change in the screen when a user clear the input 1`] = `
.c8 {
  background-color: #d9435c;
  color: #fff;
  height: 40px;
  width: 100%;
  border: 0px;
  border-radius: 4px;
  -webkit-transition: background-color 0.3s linear;
  transition: background-color 0.3s linear;
}

.c8:hover {
  background-color: #aa406d;
}

.c8:disabled {
  background-color: #ccc;
  cursor: not-allowed;
}

.c8:disabled:hover {
  background-color: #ccc;
}

.c1 {
  width: 100%;
  height: 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-image: linear-gradient( 95.02deg, #82307a 0%, #d7235d 50.36%, #e78140 100% );
}

.c2 {
  width: 90%;
  position: relative;
  color: #fffbfb;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: color 0.4s;
  transition: color 0.4s;
}

.c10 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  width: 100%;
  position: relative;
}

.c7 {
  width: 100%;
  height: 40px;
  border: 1px solid #ddd;
  border-radius: 4px;
  color: #444;
  padding: 5px 10px;
  outline: none;
}

.c7:focus {
  border: 1px solid #444;
}

.c7::-webkit-input-placeholder {
  color: #ddd;
}

.c7::-moz-placeholder {
  color: #ddd;
}

.c7:-ms-input-placeholder {
  color: #ddd;
}

.c7::placeholder {
  color: #ddd;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 20px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  padding: 50px 30px;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c11 strong {
  color: #444;
  text-align: left;
}

.c11 p {
  font-size: 13px;
  color: #ccc;
}

.c0 {
  width: 100%;
  height: 100vh;
}

.c3 {
  width: 90%;
  margin: 0 auto;
  margin-top: 35px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin: 50px 0 30px 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  gap: 10px;
  margin: 50px 0 30px 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:768px) {
  .c8 {
    width: 200px;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 72%;
    max-width: 1280px;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 72%;
    max-width: 1280px;
    margin-bottom: 70px;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        <h3>
          Online T9 converter
        </h3>
      </div>
    </header>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <input
              class="c7"
              placeholder="type a number..."
              type="text"
              value=""
            />
            
          </div>
          <button
            class="c8"
            disabled=""
            type="button"
          >
            Convert
          </button>
        </div>
        <div
          class="c9"
          data-testid="message-box"
        >
          <svg
            aria-hidden="true"
            class="c10"
            fill="#ccc"
            focusable="false"
            viewBox="0 0 24 24"
            width="60px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 2H4c-1.103 0-2 .897-2 2v12c0 1.103.897 2 2 2h3v3.767L13.277 18H20c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zm0 14h-7.277L9 18.233V16H4V4h16v12z"
            />
            <path
              d="M7 7h10v2H7zm0 4h7v2H7z"
            />
          </svg>
          <div
            class="c11"
          >
            <strong>
              Waiting for a number...
            </strong>
            <p>
              Type a number to convert its value in T9.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MainPage component Should respect the change in the screen when a user request a Post 1`] = `
.c10 {
  background-color: #d9435c;
  color: #fff;
  height: 40px;
  width: 100%;
  border: 0px;
  border-radius: 4px;
  -webkit-transition: background-color 0.3s linear;
  transition: background-color 0.3s linear;
}

.c10:hover {
  background-color: #aa406d;
}

.c10:disabled {
  background-color: #ccc;
  cursor: not-allowed;
}

.c10:disabled:hover {
  background-color: #ccc;
}

.c1 {
  width: 100%;
  height: 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-image: linear-gradient( 95.02deg, #82307a 0%, #d7235d 50.36%, #e78140 100% );
}

.c2 {
  width: 90%;
  position: relative;
  color: #fffbfb;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: color 0.4s;
  transition: color 0.4s;
}

.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  width: 100%;
  position: relative;
}

.c7 {
  width: 100%;
  height: 40px;
  border: 1px solid #ddd;
  border-radius: 4px;
  color: #444;
  padding: 5px 10px;
  outline: none;
}

.c7:focus {
  border: 1px solid #444;
}

.c7::-webkit-input-placeholder {
  color: #ddd;
}

.c7::-moz-placeholder {
  color: #ddd;
}

.c7:-ms-input-placeholder {
  color: #ddd;
}

.c7::placeholder {
  color: #ddd;
}

.c9 {
  width: 30px;
  height: auto;
  fill: #807d7d;
  position: absolute;
  left: calc(100% - 40px);
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  cursor: pointer;
  -webkit-transition: fill 0.3s linear;
  transition: fill 0.3s linear;
}

.c9:hover {
  fill: #444;
}

.c0 {
  width: 100%;
  height: 100vh;
}

.c3 {
  width: 90%;
  margin: 0 auto;
  margin-top: 35px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin: 50px 0 30px 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  gap: 10px;
  margin: 50px 0 30px 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  width: 100%;
  border: 1px solid #ccc;
  padding: 20px 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 8px;
  border-radius: 4px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-bottom: 50px;
}

@media (min-width:768px) {
  .c10 {
    width: 200px;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 72%;
    max-width: 1280px;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 72%;
    max-width: 1280px;
    margin-bottom: 70px;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        <h3>
          Online T9 converter
        </h3>
      </div>
    </header>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <input
              class="c7"
              placeholder="type a number..."
              type="text"
              value="23"
            />
            <svg
              aria-hidden="true"
              class="c8 c9"
              data-testid="clear-icon"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm75.31 260.69a16 16 0 11-22.62 22.62L256 278.63l-52.69 52.68a16 16 0 01-22.62-22.62L233.37 256l-52.68-52.69a16 16 0 0122.62-22.62L256 233.37l52.69-52.68a16 16 0 0122.62 22.62L278.63 256z"
              />
            </svg>
          </div>
          <button
            class="c10"
            type="button"
          >
            Convert
          </button>
        </div>
        <div
          class="c11"
          data-testid="converted-value-box"
        >
          <span>
            ad, 
          </span>
          <span>
            ae, 
          </span>
          <span>
            af, 
          </span>
          <span>
            bd, 
          </span>
          <span>
            be, 
          </span>
          <span>
            bf, 
          </span>
          <span>
            cd, 
          </span>
          <span>
            ce, 
          </span>
          <span>
            cf, 
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MainPage component Should respect the change in the screen when a user type a number 1`] = `
.c10 {
  background-color: #d9435c;
  color: #fff;
  height: 40px;
  width: 100%;
  border: 0px;
  border-radius: 4px;
  -webkit-transition: background-color 0.3s linear;
  transition: background-color 0.3s linear;
}

.c10:hover {
  background-color: #aa406d;
}

.c10:disabled {
  background-color: #ccc;
  cursor: not-allowed;
}

.c10:disabled:hover {
  background-color: #ccc;
}

.c1 {
  width: 100%;
  height: 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-image: linear-gradient( 95.02deg, #82307a 0%, #d7235d 50.36%, #e78140 100% );
}

.c2 {
  width: 90%;
  position: relative;
  color: #fffbfb;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: color 0.4s;
  transition: color 0.4s;
}

.c8 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  width: 100%;
  position: relative;
}

.c7 {
  width: 100%;
  height: 40px;
  border: 1px solid #ddd;
  border-radius: 4px;
  color: #444;
  padding: 5px 10px;
  outline: none;
}

.c7:focus {
  border: 1px solid #444;
}

.c7::-webkit-input-placeholder {
  color: #ddd;
}

.c7::-moz-placeholder {
  color: #ddd;
}

.c7:-ms-input-placeholder {
  color: #ddd;
}

.c7::placeholder {
  color: #ddd;
}

.c9 {
  width: 30px;
  height: auto;
  fill: #807d7d;
  position: absolute;
  left: calc(100% - 40px);
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  cursor: pointer;
  -webkit-transition: fill 0.3s linear;
  transition: fill 0.3s linear;
}

.c9:hover {
  fill: #444;
}

.c0 {
  width: 100%;
  height: 100vh;
}

.c3 {
  width: 90%;
  margin: 0 auto;
  margin-top: 35px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin: 50px 0 30px 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  gap: 10px;
  margin: 50px 0 30px 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:768px) {
  .c10 {
    width: 200px;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 72%;
    max-width: 1280px;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 72%;
    max-width: 1280px;
    margin-bottom: 70px;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        <h3>
          Online T9 converter
        </h3>
      </div>
    </header>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <input
              class="c7"
              placeholder="type a number..."
              type="text"
              value="23"
            />
            <svg
              aria-hidden="true"
              class="c8 c9"
              data-testid="clear-icon"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm75.31 260.69a16 16 0 11-22.62 22.62L256 278.63l-52.69 52.68a16 16 0 01-22.62-22.62L233.37 256l-52.68-52.69a16 16 0 0122.62-22.62L256 233.37l52.69-52.68a16 16 0 0122.62 22.62L278.63 256z"
              />
            </svg>
          </div>
          <button
            class="c10"
            type="button"
          >
            Convert
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MainPage component Should respect the initial render of page 1`] = `
.c8 {
  background-color: #d9435c;
  color: #fff;
  height: 40px;
  width: 100%;
  border: 0px;
  border-radius: 4px;
  -webkit-transition: background-color 0.3s linear;
  transition: background-color 0.3s linear;
}

.c8:hover {
  background-color: #aa406d;
}

.c8:disabled {
  background-color: #ccc;
  cursor: not-allowed;
}

.c8:disabled:hover {
  background-color: #ccc;
}

.c1 {
  width: 100%;
  height: 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-image: linear-gradient( 95.02deg, #82307a 0%, #d7235d 50.36%, #e78140 100% );
}

.c2 {
  width: 90%;
  position: relative;
  color: #fffbfb;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: color 0.4s;
  transition: color 0.4s;
}

.c10 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  width: 100%;
  position: relative;
}

.c7 {
  width: 100%;
  height: 40px;
  border: 1px solid #ddd;
  border-radius: 4px;
  color: #444;
  padding: 5px 10px;
  outline: none;
}

.c7:focus {
  border: 1px solid #444;
}

.c7::-webkit-input-placeholder {
  color: #ddd;
}

.c7::-moz-placeholder {
  color: #ddd;
}

.c7:-ms-input-placeholder {
  color: #ddd;
}

.c7::placeholder {
  color: #ddd;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 20px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  padding: 50px 30px;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c11 strong {
  color: #444;
  text-align: left;
}

.c11 p {
  font-size: 13px;
  color: #ccc;
}

.c0 {
  width: 100%;
  height: 100vh;
}

.c3 {
  width: 90%;
  margin: 0 auto;
  margin-top: 35px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin: 50px 0 30px 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  gap: 10px;
  margin: 50px 0 30px 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:768px) {
  .c8 {
    width: 200px;
  }
}

@media (min-width:768px) {
  .c2 {
    width: 72%;
    max-width: 1280px;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 72%;
    max-width: 1280px;
    margin-bottom: 70px;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        <h3>
          Online T9 converter
        </h3>
      </div>
    </header>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <input
              class="c7"
              placeholder="type a number..."
              type="text"
              value=""
            />
            
          </div>
          <button
            class="c8"
            disabled=""
            type="button"
          >
            Convert
          </button>
        </div>
        <div
          class="c9"
          data-testid="message-box"
        >
          <svg
            aria-hidden="true"
            class="c10"
            fill="#ccc"
            focusable="false"
            viewBox="0 0 24 24"
            width="60px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 2H4c-1.103 0-2 .897-2 2v12c0 1.103.897 2 2 2h3v3.767L13.277 18H20c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zm0 14h-7.277L9 18.233V16H4V4h16v12z"
            />
            <path
              d="M7 7h10v2H7zm0 4h7v2H7z"
            />
          </svg>
          <div
            class="c11"
          >
            <strong>
              Waiting for a number...
            </strong>
            <p>
              Type a number to convert its value in T9.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
